src/Build/BackEnd/Components/RequestBuilder/RequestBuilder.cs

```diff
@@ -1246,5 +1251,59 @@ private void VerifyIsNotZombie()
         {
             ErrorUtilities.VerifyThrow(!_isZombie, "RequestBuilder has been zombied.");
         }
+
+        /// <summary>
+        /// Configure warnings as messages and errors based on properties defined in the project.
+        /// </summary>
+        private void ConfigureWarningsAsErrorsAndMessages()
+        {
+            // Gather needed objects
+            //
+            ProjectInstance project = _requestEntry?.RequestConfiguration?.Project;
+            BuildEventContext buildEventContext = _projectLoggingContext?.BuildEventContext;
+            ILoggingService loggingService = _projectLoggingContext?.LoggingService;
+
+            // Ensure everything that is required is available at this time
+            //
+            if (project != null && buildEventContext != null && loggingService != null && buildEventContext.ProjectInstanceId != BuildEventContext.InvalidProjectInstanceId)
```

Comment:
Check this works on p2p node as well. Nodes initially assign a temp project ID (maybe the invalid one?) and I don't remember whether they get the global ID before or after they evaluate the project.